source/Engine.cpp

```diff
@@ -1118,6 +1130,43 @@ void Engine::Step(bool isActive)
 			GetMinablePointerColor(true),
 			3
 		});
+
+	gunsights.clear();
+	if(flagship && (flagship->Attributes().Get("gunsight") || flagship->Attributes().Get("gunsight scan power")))
+	{
+		for(shared_ptr<Ship> &ship : ships)
+		{
+			double gunsightRange = 100. * sqrt(flagship->Attributes().Get("gunsight scan power"));
+			bool isValidTarget = ship->GetSystem() == flagship->GetSystem() && !ship->IsDestroyed() && !ship->IsDisabled()
+				&& ship->IsTargetable() && !ship->Attributes().Get("inscrutable");
+			bool withinRange = ship->Position().Distance(flagship->Position()) < gunsightRange;
+			bool playerHasTarget = ship->IsYours() && (ship->GetTargetShip() || ship->GetTargetAsteroid());
+			bool isYourTarget = (flagship && ship == flagship->GetTargetShip());
+
+			if(isValidTarget && (playerHasTarget || (withinRange && isYourTarget)))
```

Comment:
Instead of using this mess of conditions, use some early ~~return~~ `continue`s.
i.e.
```cpp
if(ship->GetSystem() != flagship->GetSystem() || ship->IsDestroyed() || ship->IsDisabled()
	|| ship->IsTargetable() || ship->Attributes().Get("inscrutable"))
	continue;
```